### PR TITLE
Ms229

### PR DESCRIPTION
**Pull request by MingtaoSun**
@scoen @terenceingram @AaronCMuller @bsingh-nla 
http://ourweb.nla.gov.au/apps/jira/browse/DSCS-229
Need to get the parent PI history

### CODE REVIEW

File: src/amberdb/version/VersionedGraph.java [hunk 1]
```diff
@@ -153,12 +153,27 @@ protected void addVertexToGraph(VersionedVertex v) {
     
     
     public VersionedEdge getEdge(Long id) {
-        return getEdge(id, localMode);
+        String sql = "SELECT id, txn_start, txn_end, label, v_in, v_out, edge_order "
```
Comment (AaronCMuller) [inline src/amberdb/version/VersionedGraph.java]: does this not do the same thing as getEdge?

**Comment by scoen** [comment]
Ming i don't understand - why are you modifying this ?

File: src/amberdb/version/VersionedGraph.java (hunk 1 above)
Comment (AaronCMuller) [inline src/amberdb/version/VersionedGraph.java]: Oh I see, you've changed the params of the other getEdge

Comment (MingtaoSun) [inline src/amberdb/version/VersionedGraph.java]: I have created another method getEdgeByLabelAndOutVertexId(), so I refactored the existing one for code reusing purpose

File: src/amberdb/version/VersionedGraph.java [hunk 2]
```diff
@@ -153,12 +153,27 @@ protected void addVertexToGraph(VersionedVertex v) {
...
+    }
+    
+    public VersionedEdge getEdgeByLabelAndInVertexId(Long inVertexId, String label){
+        String sql = "SELECT id, txn_start, txn_end, label, v_in, v_out, edge_order "
+                + " FROM edge " 
+                + " WHERE v_in = :v_in_id and label = :label";
+        return getEdge(inVertexId, localMode, sql, new Object[]{inVertexId, label});
+    }
+    
+    protected VersionedEdge getEdge(Long eId, boolean localOnly, String sql, Object[] params) {
```
Comment (AaronCMuller) [inline src/amberdb/version/VersionedGraph.java]: can params not be varargs?

**Comment by MingtaoSun** [comment]
@scoen , I created another method getEdgeByLabelAndOutVertexId(). I modified the existing one to reuse the code.

**Comment by AaronCMuller** [comment]
This isn't actually required at all. The functionality that you're after is already provided by
VersionedVertex#getEdges

**Comment by scoen** [comment]
I think it'd be better to use the VersionedVertex if possible

**Pull request closed without merging**